Formula/go.rb

```diff
@@ -53,8 +53,22 @@ def install
     ENV["GOROOT_BOOTSTRAP"] = buildpath/"gobootstrap"
 
     cd "src" do
+      arch_args = []
+      if Hardware::CPU.arm?
+        ENV["GOARCH"] = "arm64"
+        # Workaround Rosetta 2 signals issues.
+        # See https://golang.org/issue/42700.
+        ENV["GODEBUG"] = "asyncpreemptoff=1"
+        arch_args = ["arch", "-x86_64"]
+      end
+
       ENV["GOROOT_FINAL"] = libexec
-      system "./make.bash", "--no-clean"
+      system *arch_args, "./make.bash", "--no-clean"
```

Comment:
```ruby
system(*arch_args, "./make.bash", "--no-clean")
```

should be sufficient to pass the syntax check

Reply:
Thank you!